src/main/scala/TableGroup.scala

```diff
@@ -1,6 +1,412 @@
+
+  def fromJson(tableGroupNode: ObjectNode, baseUri: String): TableGroup = {
+    var baseUrl = baseUri.trim
+    val containsWhitespaces = ".*\\s.*".r
+    val matcher = containsWhitespaces.pattern.matcher(baseUrl)
+    if (matcher.matches()) {
+      println(
+        "Warning: The path/url has whitespaces in it, please ensure its correctness. Proceeding with received " +
+          "path/url .."
+      )
+    }
+    val processContextResult =
+      processContextGetBaseUrlLang(tableGroupNode, baseUrl, "und")
+    baseUrl = processContextResult._1
+    val lang = processContextResult._2
+
+    restructureIfNodeIsSingleTable(tableGroupNode)
+
+    var (annotations, commonProperties, inheritedProperties, warnings) =
+      classifyPropertiesBasedOnPropertyTypeAndSetWarnings(
+        tableGroupNode,
+        baseUrl,
+        lang
+      )
+
+    val id = getId(commonProperties)
+    ensureTypeofTableGroup(tableGroupNode)
+
+    val (tables, w) = createTableObjectsAndSetWarnings(
+      tableGroupNode,
+      baseUrl,
+      lang,
+      commonProperties,
+      inheritedProperties
+    )
+    warnings = warnings.concat(w)
+
+    findForeignKeysLinkToReferencedTables(baseUrl, tables)
+
+    return TableGroup(
+      baseUrl,
+      id,
+      tables,
+      getNotes(commonProperties),
+      annotations,
+      warnings
+    )
+  }
+
+  private def restructureIfNodeIsSingleTable(
+      tableGroupNode: ObjectNode
+  ): Unit = {
+    val tableGroupNodeCopy = tableGroupNode.deepCopy()
+    if (tableGroupNode.path("tables").isMissingNode) {
+      if (!tableGroupNode.path("url").isMissingNode) {
+        val jsonArrayNode = JsonNodeFactory.instance.arrayNode()
+        jsonArrayNode.insert(0, tableGroupNodeCopy)
+        tableGroupNode.set("tables", jsonArrayNode)
+      }
+    }
+  }
+
+  def getAndValidateBaseAndLangFromContextObject(
+      contextBaseAndLangObject: ObjectNode,
+      baseUrl: String,
+      lang: String
+  ): (String, String) = {
+    var baseUrlNew = baseUrl
+    var langNew: String = lang
+
+    for ((property, value) <- contextBaseAndLangObject.getKeysAndValues) {
+      val (newValue, warning, typeString) = PropertyChecker
+        .checkProperty(property, value, baseUrl, lang)
+      if (warning.isEmpty) {
+        if (typeString == PropertyType.Context) {
+          property match {
+            case "@base"     => baseUrlNew = newValue.asText()
+            case "@language" => langNew = newValue.asText()
+          }
+        } else {
+          throw new MetadataError(
+            s"@context contains properties other than @base or @language $property)"
+          )
+        }
+      } else {
+        if (!Array[String]("@base", "@language").contains(property)) {
+          throw new MetadataError(
+            s"@context contains properties other than @base or @language $property)"
+          )
+          // Error Message class to deal with each of the warnings received
```

Comment:
Actually this comment was not addressed. I had to do what it said. Since we refactored class a lot I had to change some method signatures to pass back the warnings. Now thats done and this comment is removed.